server/errors/errors.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	te "github.com/google/trillian/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+// WrapError wraps err as a GRPC error if err is a TrillianError, else err is
+// returned unmodified.
+func WrapError(err error) error {
+	switch err := err.(type) {
+	case te.TrillianError:
+		return grpc.Errorf(codes.Code(err.Code()), err.Error())
```

Comment:
I worry about `codes.Code(err.Code())` It assumes that there's a 1:1 mapping between the two, but we've defined our own error space.  

A few ideas: 
- Define our errors in terms of the grpc codes.  This would allow us to know that the errors we've defined will map properly onto the grpc errors. 
- Define what should happen to codes we have but grpc doesn't have.